Zend/zend_portability.h

```diff
@@ -457,6 +457,12 @@ extern "C++" {
 #define ZEND_TRUTH(x)		((x) ? 1 : 0)
 #define ZEND_LOG_XOR(a, b)		(ZEND_TRUTH(a) ^ ZEND_TRUTH(b))
 
+/**
+ * Do a three-way comparison of two integers and returns -1, 0 or 1
+ * depending on whether #a is smaller, equal or larger than #b.
+ */
+#define ZEND_THREEWAY_COMPARE(a, b) ((a) == (b) ? 0 : ((a) < (b) ? -1 : 1))
```

Comment:
I think it would make sense to make this a function since `a`/`b` are used twice.

Reply:
I'd love to do that (I hate macros), but then each call would coerce the parameters into the one integer type declared in the function prototype.
(I wish everybody would just use C++ instead of plain C ;-))

Reply:
Oh I missed that this operates on different types. Thanks for the clarification.